cli/azd/pkg/project/dotnet_importer.go

```diff
@@ -272,9 +275,177 @@ func (ai *DotNetImporter) Services(
+		relativePath := svcConfig.RelativePath
+		var dOptions DockerProjectOptions
+
+		if bContainer.Build != nil {
+			defaultLanguage = ServiceLanguageDocker
+			relPath, err := filepath.Rel(p.Path, filepath.Dir(bContainer.Build.Dockerfile))
+			if err != nil {
+				return nil, err
+			}
+			relativePath = relPath
+
+			bArgs, err := evaluateArgsWithConfig(*manifest, bContainer.Build.Args)
+			if err != nil {
+				return nil, fmt.Errorf("evaluating build args for service %s: %w", name, err)
+			}
+			bArgsArray, reqEnv, err := buildArgsArrayAndEnv(*manifest, bContainer.Build.Secrets)
+			if err != nil {
+				return nil, fmt.Errorf("converting build args to array for service %s: %w", name, err)
+			}
+
+			dOptions = DockerProjectOptions{
+				Path:         bContainer.Build.Dockerfile,
+				Context:      bContainer.Build.Context,
+				BuildArgs:    mapToStringSlice(bArgs, "="),
+				BuildSecrets: bArgsArray,
+				BuildEnv:     reqEnv,
+			}
+		}
+
+		svc := &ServiceConfig{
+			RelativePath: relativePath,
+			Language:     defaultLanguage,
+			Host:         DotNetContainerAppTarget,
+			Docker:       dOptions,
+		}
+
+		svc.Name = name
+		svc.Project = p
+		svc.EventDispatcher = ext.NewEventDispatcher[ServiceLifecycleEventArgs]()
+
+		svc.Infra.Provider, err = provisioning.ParseProvider(svc.Infra.Provider)
+		if err != nil {
+			return nil, fmt.Errorf("parsing service %s: %w", svc.Name, err)
+		}
+
+		svc.DotNetContainerApp = &DotNetContainerAppOptions{
+			ContainerImage: bContainer.Image,
+			Manifest:       manifest,
+			ProjectName:    name,
+			AppHostPath:    svcConfig.Path(),
+		}
+		services[svc.Name] = svc
+
+	}
 	return services, nil
 }
 
+// buildArgsArray produces an array of args to pass to the container build command.
+// See: https://docs.docker.com/build/building/secrets/
+func buildArgsArrayAndEnv(
+	manifest apphost.Manifest,
+	bArgs map[string]apphost.ContainerV1BuildSecrets) ([]string, []string, error) {
+	var result []string
+	var reqEnv []string
+
+	for bArgKey, bArg := range bArgs {
+		if bArg.Type != "env" && bArg.Type != "file" {
+			return nil, nil, fmt.Errorf("unsupported secret type %q for build arg %q", bArg.Type, bArgKey)
+		}
+
+		baseArg := fmt.Sprintf("id=%s", bArgKey)
+		if bArg.Type == "file" {
+			if bArg.Source == nil {
+				return nil, nil, fmt.Errorf("missing source for file secret %q", bArgKey)
+			}
+			baseArg = fmt.Sprintf("id=%s,src=%s", bArgKey, *bArg.Source)
+		}
+		if bArg.Type == "env" {
+			if bArg.Value == nil {
+				return nil, nil, fmt.Errorf("missing value for env secret %q", bArgKey)
+			}
+			bArgValue, err := evaluateExpressions(*bArg.Value, manifest)
+			if err != nil {
+				return nil, nil, fmt.Errorf("evaluating value for env secret %q: %w", bArgKey, err)
+			}
+			reqEnv = append(reqEnv, fmt.Sprintf("%s=%s", bArgKey, bArgValue))
+		}
+		result = append(result, baseArg)
+	}
+
+	return result, reqEnv, nil
+}
+
+var argExpression = regexp.MustCompile(`\{([^}]+)\}`)
```

Comment:
We were not able to leverage `apphost.EvalString` in some way here?  I feel like it should be flexible enough to do what you want, but maybe I'm wrong.  Sort of feels like we are duplicating some of it's logic here via regex's and I'd love to not do that if possible.

Reply:
There are a few things that I prefer about using a regex instead of a EvalString, for example:
- EvalString expects any `{` to have a corresponding `}` or fails. You need to scape `{{` to ignore. This logic works for how the apphost creates expressions for templates, but it does not directly reflect the logic from the manifest, where I  don't think there will be ever a `{{`.

But if you have a preference for EvalString, I'll make it happen.

(If you ask me, I would change apphost.EvalString for regext.ReplaceAllStringFn() ) =P